plugins/modules/fusion_info.py

```diff
@@ -81,260 +81,459 @@
-    default_info["users"] = len(users)
+    return inner
 
-    host_access_policies = host_access_api_instance.list_host_access_policies()
-    default_info["host_access_policies"] = len(host_access_policies.items)
 
-    hw_types = hw_api_instance.list_hardware_types()
-    default_info["hardware_types"] = len(hw_types.items)
+def generate_default_dict(module, fusion):
+    def warning_api_exception(name):
+        module.warn(f"Cannot get {name} in [default dict], reason: Permission denied")
 
-    tenants = tenant_api_instance.list_tenants()
-    default_info["tenants"] = len(tenants.items)
-    tenant_spaces = 0
-    for tenant in range(0, len(tenants.items)):
-        tenant_spaces = tenant_spaces + len(
-            tenantspace_api_instance.list_tenant_spaces(
-                tenant_name=tenants.items[tenant].name
-            ).items
+    def warning_argument_none(name, requirement):
+        module.warn(
+            f"Cannot get {name} in [default dict], reason: Required argument `{requirement}` not available."
         )
-    default_info["tenant_spaces"] = tenant_spaces
-
-    roles = roles_api_instance.list_roles()
-    assignments = 0
-    default_info["roles"] = len(roles)
-    for role in range(0, len(roles)):
-        assignments = assignments + len(
-            role_assign_api_instance.list_role_assignments(role_name=roles[role].name)
+
+    # All values are independent on each other - if getting one value fails, we will show warning and continue.
+    # That's also the reason why there's so many nested for loops repeating all over again.
+    version = None
+    users_num = None
+    protection_policies_num = None
+    host_access_policies_num = None
+    hardware_types_num = None
+    storage_services = None
+    storage_services_num = None
+    tenants = None
+    tenants_num = None
+    regions = None
+    regions_num = None
+    roles = None
+    roles_num = None
+    storage_classes_num = None
+    role_assignments_num = None
+    tenant_spaces_num = None
+    volumes_num = None
+    placements_groups_num = None
+    snapshots_num = None
+    availability_zones_num = None
+    appliances_num = None
+    network_interfaces_num = None
+    network_interface_groups_num = None
+    storage_endpoints_num = None
+
+    try:
+        version = purefusion.DefaultApi(fusion).get_version().version
+    except purefusion.rest.ApiException as exc:
+        if exc.status == 403:
+            warning_api_exception("API version")
+
+    try:
+        users_num = len(purefusion.IdentityManagerApi(fusion).list_users())
+    except purefusion.rest.ApiException as exc:
+        if exc.status == 403:
+            warning_api_exception("Users")
+
+    try:
+        protection_policies_num = len(
+            purefusion.ProtectionPoliciesApi(fusion).list_protection_policies().items
         )
-    default_info["role_assignments"] = assignments
+    except purefusion.rest.ApiException as exc:
+        if exc.status == 403:
```

Comment:
Please use a code from [http](https://docs.python.org/3/library/http.html#module-http)
In other places as well.